jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py

```diff
@@ -48,6 +48,27 @@ def _post_install_packages( self ):
         self.__install_parasol( )
         self.__patch_distutils( )
         self.__configure_gridengine( )
+        self.__createMesosCredentialsFile( )
+        self.__setEphemeralTempDir( )
+
+    @fabric_task
+    def __createMesosCredentialsFile(self):
+        # Create the credentials file and transfer ownership to mesosbox
+        sudo( 'echo toil liot > /var/lib/mesos/credentials.txt' )
+        sudo( 'chown jenkins:jenkins /var/lib/mesos/credentials.txt')
+        sudo( 'chmod 755 /var/lib/mesos/credentials.txt')
+
+    @fabric_task
+    def __setEphemeralTempDir(self):
+        """
+        This sets the value of the environment variable TMPDIR to one located on /mnt/ephemeral.
+        This way, all temp dirs created for tests will go to /mnt/ephemeral where there is a lot
+        more disk space to carry out tests.
+        """
+        sudo( 'mkdir /mnt/ephemeral/temp' )
+        sudo( 'chown jenkins:jenkins /mnt/ephemeral/temp' )
+        sudo( 'chmod 755 /mnt/ephemeral/temp' )
+        sudo( 'echo TMPDIR=/mnt/ephemeral/temp > /etc/environment')
```

Comment:
This won't work unfortunately. The change to /etc/environment will be persisted to the image (if one is made of the box). The creation of /mnt/ephemeral/temp will not. When a box is recreated from an image, it will likely fail because TMPDIR refers to an empty directory.

We get a similar effect by doing 

```
mkdir -p /mnt/ephemeral/tmp && export TMPDIR=/mnt/ephemeral/tmp
```

before every build in Jenkins. That's easy to do. In fact, I will do that right now for the Toil builds.

In the long run, having /tmp on ephemeral is generally beneficial, for all cgcloud boxes. It would have to be done in the same place where /mnt/ephemeral is prepared, CloudInitBox I think. Also, a bind mount would be better than overriding TMPDIR. That way, the logical path would remain /tmp while physically, things would end up in /mnt/ephemeral/tmp. We'd have to think about files already present in /tmp.